### PR TITLE
report: celebrate your all 100's

### PR DESCRIPTION
**Pull request by paulirish**
It's come up before. But it came up today: https://twitter.com/RobertJGabriel/status/1004604294102945793

The logic is, reward within the report if all the categories in your report have a 100. This can trigger even if you're only testing a single category.  I considered enforcing that 2 or more categories are present, but that seems arbitrary. 

(A further thing to explore: add additional fireworks per each included category. From a trickle up to a finale.)

example report: 

* https://ripe-church.surge.sh/score100.html
* https://ripe-church.surge.sh/solocat100.html

### CODE REVIEW

**Comment by JoelEinbinder** [comment]
```css
.score100 .lh-lighthouse {
    animation: 4s linear infinite party-lighthouse;
}

@keyframes party-lighthouse {
    from{
        filter: hue-rotate(0deg);
    }
    to{
        filter: hue-rotate(360deg);
    }
}
```

**Comment by patrickhulce** [comment]
Ha, this is awesome!!! ❤️

**Comment by patrickhulce** [comment]
@hwikyounglee and I were just thinking it might be cool to go all out and replace the dark gray background and do a animation through the rainbow :D 

```css
.score100 .lh-header-bg {
  background: #f00;
  animation: 4s linear infinite party-background;
}

@keyframes party-background {
    from{
        filter: hue-rotate(0deg);
    }
    to{
        filter: hue-rotate(360deg);
    }
}
```

**Comment by hwikyounglee** [comment]
A suggestion to the excitement: shall we have a way to stop the animation to provide a control?
Options might be: play only once on load, stop when click the sky, stop on scrolling, or a combination of those.

**Comment by paulirish** [comment]
see commits for changes

* https://ripe-church.surge.sh/score100.html
* https://ripe-church.surge.sh/solocat100.html

**Comment by benschwarz** [comment]
My vote is for lights coming out of he ligthouse as if there’s an epic party going on inside 👏

**Comment by hwikyounglee** [comment]
- For the collapsed header, either the text might need to be inverted or go back to the regular gray bar?  
image
- I feel we should go with **all** category 100's for fireworks. For a solo cat 100, we could consider a confetti in each gauge in the future perhaps? 
- Also, wondering about the battery usage of the animation: if it consumes battery, I suggest the animation repeats x times (x=3~5?). 

Thanks Paul!

**Comment by ebidel** [comment]
Lighthouse blue!? It's safer to sail in the day time.

screen shot 2018-06-08 at 1 06 40 pm

Also noticed a few scroll bugs. 

Background area shows through header:

screen shot 2018-06-08 at 1 06 45 pm

Is the share icon meant to be see through?

screen shot 2018-06-08 at 1 07 13 pm

As @hwikyounglee pointed this out too. The text is hard to see:

screen shot 2018-06-08 at 1 06 49 pm

**Comment by paulirish** [comment]
examples are updated:

* https://ripe-church.surge.sh/score100.html
* https://ripe-church.surge.sh/solocat100.html

**Comment by exterkamp** [comment]
Took @JoelEinbinder's amazing party CSS and applied it to just the lighthouse light:
fe8fffb5-1a62-479e-b90f-f08752c56cb4
Yes? or very yes?

**Comment by patrickhulce** [comment]
oh VERY yes!

File: lighthouse-core/report/html/renderer/report-ui-features.js [hunk 1]
```diff
@@ -241,10 +241,12 @@ class ReportUIFeatures {
     scoreShadow.style.transform = `scaleY(${1 - scrollPct * 0.2})`;
 
     // Fade & move the scorescale
-    const scoreScalePositionDelta = 32;
-    const scoreScale = this._dom.find('.lh-scorescale', scoresContainer);
-    scoreScale.style.opacity = `${1 - scrollPct}`;
-    scoreScale.style.transform = `translateY(${scrollPct * -scoreScalePositionDelta}px)`;
+    try {
```
Comment (paulirish) [inline lighthouse-core/report/html/renderer/report-ui-features.js]: I think we don't need this try/catch any longer.. pretty sure it was resolved somewhere else.. but IIRC it only applied in the single-category case.

**Comment by exterkamp** [comment]
Updated examples:
- [full](http://simplistic-ghost.surge.sh/example.com_2019-01-10_10-45-43.report.html)
- [one category](http://simplistic-ghost.surge.sh/example.com_2019-01-10_10-47-43.report)

File: lighthouse-core/report/html/renderer/report-ui-features.js (hunk 1 above)
Comment (exterkamp) [reply lighthouse-core/report/html/renderer/report-ui-features.js]: I think we still need it.  Even on master it throws errors when scrolling in single category reports.  I can split this into another PR if it doesn't belong here, but I think this is still a bug.
master:
image

**Comment by Hoten** [comment]
Super fun I want this yesterday.

just gonna bring up @hwikyounglee's concerns about how this could be distracting.

> A suggestion to the excitement: shall we have a way to stop the animation to provide a control?
> Options might be: play only once on load, stop when click the sky, stop on scrolling, or a combination of those.

WDYT of stopping when clicking?

File: lighthouse-core/report/html/renderer/report-ui-features.js (hunk 1 above)
Comment (paulirish) [reply lighthouse-core/report/html/renderer/report-ui-features.js]: lets just move a few lines inside of a 

```js
const scoreScale = this._dom.find('.lh-scorescale', scoresContainer);
if (scoreScale) { ... }
```

**Comment by paulirish** [comment]
Personally, I dont think it's worth adding user control to stop the animation. But Shane already did it so whatever. :)

File: lighthouse-core/report/html/renderer/report-ui-features.js (hunk 1 above)
Comment (exterkamp) [reply lighthouse-core/report/html/renderer/report-ui-features.js]: The error is on the find itself.  This still throws, no?

Comment (paulirish) [reply lighthouse-core/report/html/renderer/report-ui-features.js]: what i meant to say is.. "what a beautiful try/catch! let's keep it just like it is."

**Comment by exterkamp** [comment]
examples:
- http://unable-fruit.surge.sh/one_category
- http://unable-fruit.surge.sh/all_categories

**Comment by paulirish** [comment]
🎆 🎆 🎆 🎆 🎆 🎆 🎆 🎆 🎆 🎆 🎆 🎆 🎆 🎆 🎆 🎆 🎆 🎆 🎆 🎆

**Pull request merged**